### PR TITLE
Revert "fix: allow specifying a base for source maps"

### PR DESCRIPTION
**Pull request by dsherret**
Temporarily reverting this one because it causes source maps to have blank source file names and there's too much to deal with WRT the latest swc upgrade (so let's do a follow up PR).

image

Reverts denoland/deno_ast#87

### CODE REVIEW

File: src/transpiling/mod.rs [hunk 1]
```diff
@@ -138,21 +138,11 @@ impl EmitOptions {
...
-  pub maybe_base: Option<ModuleSpecifier>,
 }
 
 impl crate::swc::common::source_map::SourceMapGenConfig for SourceMapConfig {
   fn file_name_to_source(&self, f: &FileName) -> String {
-    match f {
-      FileName::Url(specifier) => self
-        .maybe_base
-        .as_ref()
-        .and_then(|base| base.make_relative(specifier))
```
Comment (dsherret) [inline src/transpiling/mod.rs:152]: This does `"file:///file.js".make_relative("file:///file.js") === ""`

**Review by bartlomieju — APPROVED** [review]

**Comment by kitsonk** [comment]
I think the base shouldn't generally be the same file as a file in the source map.  Was there a specific scenario where the screen grab was from?

**Comment by dsherret** [comment]
@kitsonk yeah, this is v2::tests::import_map in eszip

It seems they are the same because of this line?

https://github.com/denoland/deno_ast/pull/92/files#diff-c865a1cff44ff278a457dcd63af4492767b0f67655e139b489dd961b5c1b8e3eL182

**Comment by kitsonk** [comment]
Ok, I think the fix would be to guard against equality and to just output the basename in such cases.  I guess eszip keeps the same files indexed by their original filename versus an output filename.  Should be a straight forward fix.

File: src/transpiling/mod.rs (hunk 1 above)
Comment (dsherret) [reply src/transpiling/mod.rs:152]: I don't understand why someone was providing a base that was a filename. This code looks perfectly fine to me.

**Pull request merged**